packages/agent/src/agents/operators/mcp-agent/mcpAgentTools.ts

```
@@ -0,0 +1,585 @@
+          const currentMcpServers = agent.mcpServers || {};
+
+          const finalServerName =
+            serverName || qualifiedName.split('/').pop() || qualifiedName;
+
+          if (currentMcpServers[finalServerName]) {
+            throw new Error(
+              `MCP server "${finalServerName}" already exists for agent "${agentId}"`
+            );
+          }
+
+          const smitheryConfig: any = {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)